syne_tune/experiments/plotting.py

```diff
@@ -434,20 +435,83 @@ def _figure_shape(plot_params: PlotParameters) -> Tuple[int, int]:
             nrows = ncols = 1
         return nrows, ncols
 
-    def _aggregrate_results(
+    def _enrich_results_per_experiment(
+        self,
+        df: pd.DataFrame,
+        column_name: str,
+        dataframe_column_generator: Optional[DataFrameColumnGenerator],
+    ) -> pd.DataFrame:
+        if dataframe_column_generator is None:
+            return df
+        assert column_name not in df.columns, (
+            f"New column to be appended to results dataframe: {column_name} is "
+            f"already a column: {df.columns}"
+        )
+        new_column = dataframe_column_generator(df)
+        return df.assign(**{column_name: new_column})
+
+    def _extract_result_curves_per_experiment(
+        self,
+        df: pd.DataFrame,
+        plot_params: PlotParameters,
+        extra_results_keys: Optional[List[str]],
+        one_trial_special: bool,
+        setup_name: str,
+        subplot_no: int,
+        extra_results: Dict[str, Any],
+        prev_max_rt: Optional[float],
+        xlim: Tuple[float, float],
+    ) -> Dict[str, Any]:
+        metric = plot_params.metric
+        show_init_trials = plot_params.show_init_trials
+        metric_multiplier = plot_params.metric_multiplier
+        if one_trial_special:
+            # Filter down the dataframe
+            df = df[df["trial_id"] <= show_init_trials.trial_id]
+        if plot_params.mode == "max":
+            ys = metric_multiplier * np.array(df[metric].cummax())
+            if plot_params.convert_to_min:
+                ys = 1 - ys
+        else:
+            ys = metric_multiplier * np.array(df[metric].cummin())
+        rt = np.array(df[ST_TUNER_TIME])
+        if one_trial_special:
+            # Extend curve to the end, so it can be seen
+            rt = np.append(rt, prev_max_rt)
+            ys = np.append(ys, ys[-1])
+        if xlim is not None:
+            # Slice w.r.t. time. Doing this here, speeds up
+            # aggregation
+            ind = np.logical_and(rt >= xlim[0], rt <= xlim[1])
+            rt = rt[ind]
+            ys = ys[ind]
+        # Collect extra results
+        if extra_results_keys is not None and not one_trial_special:
+            if self._metadata_subplot_level:
+                key_sequence = [setup_name, subplot_no]
+            else:
+                key_sequence = [setup_name]
+            final_pos = df[ST_TUNER_TIME].idxmax()
+            final_row = dict(df.loc[final_pos])
+            for key in extra_results_keys:
+                _insert_into_nested_dict(
+                    dictionary=extra_results,
+                    key_sequence=key_sequence + [key],
+                    value=final_row[key],
+                )
```

Comment:
I am not sure if I understand this part. Is this modifying the `extra_results` input argument? I think it woukd be good to add a docstring ot the function to make it clear that this is modified.